array.go

```diff
@@ -640,7 +640,7 @@ func (a *arrayObject) MemUsage(ctx *MemUsageContext) (memUsage uint64, err error
 			return memUsage, err
 		}
 
-		ctx.Ascend()
```

Comment:
Thank you for tackling this! Sorry for lacking more details on the ticket.

The reason @jwongmongodb was suggesting a `defer` is so that I wouldn't need to call `ctx.Ascend()` right before every return statement. That's pretty much what `defer` does! In go it is idiomatic to couple the `defer` calls to whatever they are related to (more commonly this happens when you acquire a resource and want to make sure it gets released upon the function's return). In this case it would be `ctx.Descend()`.

(You can read a bit more about `defer` [here](https://go.dev/blog/defer-panic-and-recover))

```go
    if err := ctx.Descend(); err != nil {
		return memUsage, err
	}
    defer ctx.Acend()
```

With this change you can remove any call to `ctx.Ascend()` we had before.

Reply:
This applies to all the other files as well, I just left a comment on the first one.

Reply:
Ohh got it thank you sm! So sorry for the misunderstanding still getting comfortable with Go 😅